Reproducible Failure in org.elasticsearch.xpack.security.SecurityInfoTransportActionTests.testUsage

This fails 100% of the time for me locally on `master`:

```
REPRODUCE WITH: ./gradlew ':x-pack:plugin:security:test' --tests "org.elasticsearch.xpack.security.SecurityInfoTransportActionTests.testUsage" -Dtests.seed=372C46A9DB6C1F34 -Dtests.locale=en-US -Dtests.timezone=Etc/GMT-14 -Druntime.java=11

Suite: Test class org.elasticsearch.xpack.security.SecurityInfoTransportActionTests
  1> [2021-09-22T21:33:22,526][INFO ][o.e.x.s.SecurityInfoTransportActionTests] [testUsage] before test
  1> [2021-09-22T21:33:22,774][INFO ][o.e.x.s.SecurityInfoTransportActionTests] [testUsage] after test
  2> REPRODUCE WITH: ./gradlew ':x-pack:plugin:security:test' --tests "org.elasticsearch.xpack.security.SecurityInfoTransportActionTests.testUsage" -Dtests.seed=372C46A9DB6C1F34 -Dtests.locale=en-US -Dtests.timezone=Etc/GMT-14 -Druntime.java=11
  2> java.lang.AssertionError: 
    Expected: is <false>
         but: was <true>
        at __randomizedtesting.SeedInfo.seed([372C46A9DB6C1F34:8B8BDA7D6710C7BD]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.xpack.security.SecurityInfoTransportActionTests.testUsage(SecurityInfoTransportActionTests.java:193)
  2> NOTE: leaving temporary files on disk at: /home/brownbear/src/elasticsearch/x-pack/plugin/security/build/testrun/test/temp/org.elasticsearch.xpack.security.SecurityInfoTransportActionTests_372C46A9DB6C1F34-002
  2> NOTE: test params are: codec=Asserting(Lucene90): {}, docValues:{}, maxPointsInLeafNode=27, maxMBSortInHeap=7.900295912426337, sim=Asserting(RandomSimilarity(queryNorm=false): {}), locale=en-US, timezone=Etc/GMT-14
  2> NOTE: Linux 5.11.0-36-generic amd64/Oracle Corporation 11.0.2 (64-bit)/cpus=48,threads=1,free=511214080,total=536870912
  2> NOTE: All tests run in this JVM: [SecurityInfoTransportActionTests]

Tests with failures:
 - org.elasticsearch.xpack.security.SecurityInfoTransportActionTests.testUsage

```

--- a/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/SecurityInfoTransportActionTests.java
+++ b/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/SecurityInfoTransportActionTests.java
@@ -117,7 +117,7 @@ public class SecurityInfoTransportActionTests extends ESTestCase {
             apiKeyServiceEnabled = randomBoolean();
             settings.put("xpack.security.authc.api_key.enabled", apiKeyServiceEnabled);
         } else {
-            apiKeyServiceEnabled = httpSSLEnabled;
+            apiKeyServiceEnabled = true; // this is the default
         }
 
         final boolean auditingEnabled = randomBoolean();

